package/MDAnalysis/lib/util.py

```diff
@@ -1267,6 +1267,54 @@ def fixedwidth_bins(delta, xmin, xmax):
     dx = 0.5 * (N * _delta - _length)  # add half of the excess to each end
     return {'Nbins': N, 'delta': _delta, 'min': _xmin - dx, 'max': _xmax + dx}
 
+def get_weights(atoms, weights):
+    """Check that a `weights` argument is compatible with `atoms`.
+
+    Parameters
+    ----------
+    atoms : AtomGroup or array_like
+        The atoms that the `weights` should be applied to. Typically this
+        is a :class:`AtomGroup` but because only the length is compared,
+        any sequence for which ``len(atoms)`` is defined is acceptable.
+    weights : {"mass", None} or array_like
+        All MDAnalysis functions or classes understand "mass" and will then
+        use ``atoms.masses``. ``None`` indicates equal weights for all atoms.
+        Using an ``array_like`` assigns a custom weight to each element of
+        `atoms`.
+
+    Returns
+    -------
+    weights : array_like or None
+         If "mass" was selected, ``atoms.masses`` is returned, otherwise the
+         value of `weights` (which can be ``None``).
+
+    Raises
+    ------
+    TypeError
+        If `weights` is not one of the allowed values or if it is not a 1D
+        array with the same length as `atoms`, then the exception is raised.
+        :exc:`TypeError` is also raised if ``atoms.masses`` is not defined.
+    """
+    if weights == "mass":
+        try:
+            weights = atoms.masses
```

Comment:
Using `atoms.masses` breaks the idea that *anything in `lib.util` is independent from AtomGroup* etc. However, I didn't really know of a better place to put it. Furthermore, one *could* get around it with duck-typing `atoms` (or just catching the `TypeError`).

Anyway, suggestions/comments welcome.